key/src/main/java/software/amazon/kms/key/ModelAdapter.java

```diff
@@ -1,23 +1,30 @@
 package software.amazon.kms.key;
 
 import com.amazonaws.util.StringUtils;
+
+import software.amazon.awssdk.services.kms.model.CustomerMasterKeySpec;
 import software.amazon.awssdk.services.kms.model.KeyUsageType;
 
 public class ModelAdapter {
-    protected static final String DEFAULT_DESCRIPTION = "";
-    protected static final Boolean DEFAULT_ENABLED = true;
-    protected static final Boolean DEFAULT_ENABLE_KEY_ROTATION = false;
+    private static final String DEFAULT_DESCRIPTION = "";
+    private static final Boolean DEFAULT_ENABLED = true;
+    private static final Boolean DEFAULT_ENABLE_KEY_ROTATION = false;
+    private static final String DEFAULT_KEY_USAGE = KeyUsageType.ENCRYPT_DECRYPT.toString();
+    private static final String DEFAULT_KEY_SPEC = CustomerMasterKeySpec.SYMMETRIC_DEFAULT.toString();
```

Comment:
so it means if nothing was specified but `KeyPolicy` then we create plain vanilla symmetric key?

Reply:
Yes, if only a KeyPolicy is specified, we create a symmetric key. This is necessary to maintain backwards compatibility with older templates. In the current state of the world, creating a key with only a KeyPolicy also creates a symmetric key.